extensions/gdx-setup/src/com/badlogic/gdx/setup/ExternalExtensionsDialog.java

```diff
@@ -0,0 +1,244 @@
+                addToDependencyMapFromXML(dependencies, eElement, "android");
+                addToDependencyMapFromXML(dependencies, eElement, "ios");
+                addToDependencyMapFromXML(dependencies, eElement, "html5");
+
+                ExtensionRowCheckbox checkBox = new ExtensionRowCheckbox(rowText);
+                checkBoxeMap.put(name, checkBox);
+
+                content.add(checkBox, new GridBagConstraints(0, i, 1, 1, 1, 1, NORTH, HORIZONTAL, new Insets(0, 0, 0, 0), 0, 0));
+
+                final ExternalExtension extension = new ExternalExtension(name, description, version);
+                extension.setDependencies(dependencies);
+
+                checkBox.addItemListener(new ItemListener() {
+                    @Override
+                    public void itemStateChanged(ItemEvent e) {
+                        ExtensionRowCheckbox box = (ExtensionRowCheckbox) e.getSource();
+                        if (box.isSelected()) {
+                            extensions.add(extension);
+                        } else {
+                            if (extensions.contains(extension)) {
+                                extensions.remove(extension);
+                            }
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    private void uiStyle() {
+        contentPane.setBackground(new Color(36, 36, 36));
+
+        topPanel.setBackground(new Color(36, 36, 36));
+        topPanel.setForeground(new Color(255, 255, 255));
+        content.setBackground(new Color(36, 36, 36));
+        content.setForeground(new Color(255, 255, 255));
+        bottomPanel.setBackground(new Color(36, 36, 36));
+        bottomPanel.setForeground(new Color(255, 255, 255));
+        buttonPanel.setBackground(new Color(36, 36, 36));
+        buttonPanel.setForeground(new Color(255, 255, 255));
+
+        warningNotice.setForeground(new Color(200, 20, 20));
+    }
+
+    void onOK () {
+        setVisible(false);
+    }
+
+    void onCancel () {
+        setVisible(false);
+        restore();
+    }
+
+    private void takeSnapshot () {
+        extensionsSnapshot.clear();
+        for(int i = 0; i < extensions.size(); i++) {
+            extensionsSnapshot.add(extensions.get(i));
+        }
+    }
+
+    private void restore() {
+        extensions.clear();
+        for (ExtensionRowCheckbox checkBox : checkBoxeMap.values()) {
+            checkBox.setSelected(false);
+        }
+        for(int i = 0; i < extensionsSnapshot.size(); i++) {
+            extensions.add(extensionsSnapshot.get(i));
+            checkBoxeMap.get(extensionsSnapshot.get(i).getName()).setSelected(true);
+        }
+    }
+
+    private void addToDependencyMapFromXML(HashMap<String, String> dependencies, Element eElement, String platform) {
+        if(eElement.getElementsByTagName(platform).item(0) != null) {
+            dependencies.put(platform, eElement.getElementsByTagName(platform).item(0).getTextContent());
```

Comment:
Project dependencies can also be null if they are not supported by that platform.